src/pages/GetStarted/CustomWorkspaceTab/DevfileSelector/DevfileSelector.tsx

```diff
@@ -57,41 +63,52 @@ export class DevfileSelectorFormGroup extends React.PureComponent<Props, State>
     this.devfileLocationRef = React.createRef();
   }
 
-  private onDevfileSelected(meta: che.DevfileMetaData): void {
+  private handleDevfileClear(): void {
+    if (this.props.onClear) {
+      this.props.onClear();
+    }
+  }
+
+  private async handleDevfileSelect(meta: che.DevfileMetaData): Promise<void> {
     // clear location input
     this.devfileLocationRef.current?.clearInput();
-
     try {
-      this.loadDevfile(meta.links.self);
-    } catch {
-      // noop
+      const devfileContent = await this.props.requestDevfile(meta.links.self);
+      const devfile = safeLoad(devfileContent);
+      this.props.onDevfile(devfile);
+    } catch (e) {
+      let errorMessage = 'Failed to load devfile.';
+      console.warn(errorMessage, e);
+      if (e) {
+        errorMessage = `${errorMessage} ${e}`;
+      }
+      this.showAlert({
+        key: 'load-devfile-failed',
+        title: errorMessage,
+        variant: AlertVariant.warning,
+      });
+      throw new Error(e);
     }
   }
 
   private async onLocationChanged(location: string): Promise<void> {
     // clear devfile select
     this.devfileSelectRef.current?.clearSelect();
-
-    try {
-      await this.loadDevfile(location);
-    } catch {
-      this.devfileLocationRef.current?.invalidateInput();
-    }
-  }
-
-  private async loadDevfile(location: string): Promise<void> {
     try {
-      const devfileContent = await this.props.requestDevfile(location);
+      const devfileContent = await this.props.requestFactoryResolver(location);
       this.props.onDevfile(devfileContent);
     } catch (e) {
-      console.warn('Failed to load devfile.', e);
-
+      this.devfileLocationRef.current?.invalidateInput();
+      let errorMessage = 'Failed to load factory resolver.';
```

Comment:
I'd like to reword this message to be `Failed to resolve or load the devfile.`

Reply:
OK. Done